### ISSUE TITLE
Cannot override `disallow_subclassing_any` for `matplotlib.*` in `pyproject.toml`

### ISSUE THREAD

**Issue by bersbersbers** [python/mypy#15250]
**Bug Report**

I cannot override `disallow_subclassing_any` for `matplotlib.*` in `pyproject.toml`

**To Reproduce**

```python
"""False-positive 'Class cannot subclass'."""
from matplotlib.backends.backend_qtagg import FigureCanvasQTAgg

class Class(FigureCanvasQTAgg):
    """Class."""
```

**Expected Behavior**

No error

**Actual Behavior**

```
bug.py:4: error: Class cannot subclass "FigureCanvasQTAgg" (has type "Any")  [misc]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: mypy 1.3.0 (compiled: yes)
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): see below
- Python version used: 3.11.3

`pyproject.toml`
```python
[tool.mypy]
ignore_missing_imports = false  # this works
disallow_subclassing_any = true # this works

[[tool.mypy.overrides]]
module = "matplotlib.*"
ignore_missing_imports = true    # this works
disallow_subclassing_any = false # this does not
```
